test/model/card/action/card_action_test.dart

```diff
@@ -0,0 +1,47 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+import 'package:treasure_of_the_high_seas/model/card/action/card_action.dart';
+import 'package:treasure_of_the_high_seas/model/game_state.dart';
+import 'package:treasure_of_the_high_seas/model/resource.dart';
+
+import '../../../mocks.dart';
+import '../../../test_utils.dart';
+
+main() {
+  test('should be enabled if cost is affordable', () {
+    final state = makeGameState(playerHand: [Resource.CREW]);
+
+    final affordable = new _FakeAction([Resource.CREW]);
+    final unaffordable = new _FakeAction([Resource.LANDLUBBER]);
+
+    expect(affordable.isEnabled(state), true);
+    expect(unaffordable.isEnabled(state), false);
+  });
+
+  test('should deduct the cost, perform the action then move on to the next card', () {
+    final state = makeGameState(playerHand: [Resource.CREW, Resource.LANDLUBBER]);
+    state.nextCard();
+
+    final fn = new MockFunction().fn;
```

Comment:
Open to suggestions on this as it's pretty gross. I basically want to verify that `performActionImpl` gets called. I suppose I could give `FakeAction` a boolean member and update it to be true, then verify that's happened... but that's also gross.

You [can't mock a function](https://github.com/dart-lang/mockito/issues/62) easily... and I can't make `_FakeAction` extend `Mock` because of single inheritance. Sigh...

Reply:
This does look like hard work for a simple test, although I don't have a better suggestion since I have yet to write a single test in dart!